tools/gitignore/gitignore.py

```diff
@@ -17,6 +17,12 @@ def fnmatch_translate(pat, path_name=False):
     else:
         any_char = "."
         parts.append("^(?:.*/)?")
+    if pat[-1] == "/":
```

Comment:
I think parse_line doesn't pass the trailing `/` so we might need a workaround that condition first.
Also, afaict fnmatch_translate _does_ have a condition to deal with directories but due to parse_line, we never enter that condition. Not sure about the correctness of the regexp fnmatch came with for the directories though.